lib/rex/post/meterpreter/ui/console/command_dispatcher/sniffer.rb

```diff
@@ -28,14 +28,26 @@ def initialize(shell)
   # List of supported commands.
   #
   def commands
+    #all = {
     {
       "sniffer_interfaces" => "Enumerate all sniffable network interfaces",
       "sniffer_start" => "Start packet capture on a specific interface",
       "sniffer_stop"  => "Stop packet capture on a specific interface",
       "sniffer_stats" => "View statistics of an active capture",
       "sniffer_dump"  => "Retrieve captured packet data to PCAP file",
-      "sniffer_release" => "Free captured packets on a specific interface instead of downloading them",
+      "sniffer_release" => "Free captured packets on a specific interface instead of downloading them"
     }
+
+    #reqs = {
+    #  "sniffer_interfaces" => ['sniffer_interfaces'],
+    #  "sniffer_start" => ['sniffer_capture_start'],
+    #  "sniffer_stop"  => ['sniffer_capture_stop'],
+    #  "sniffer_stats" => ['sniffer_capture_stats'],
+    #  "sniffer_dump"  => ['sniffer_capture_dump'],
+    #  "sniffer_release" => ['sniffer_capture_release']
+    #}
+
+    #filter_commands(all, reqs)
```

Comment:
This code didn't work as-expected when I originally tested it (like it showed the sniffer_XXX commands even if they hadn't been loaded by Mettle), I'll give it another go and remove if it's still funky.

Reply:
It looks like get_loaded_extension_commands is called before core_loadlib?

Reply:
I don't think the sniffer commands show at all with this code in place. core_enumext commands is called before core_loadlib and core_loadlib isn't returning the sniffer commands currently.
hacky fix:
```
diff --git a/mettle/src/coreapi.c b/mettle/src/coreapi.c
index be732cf..0f59379 100644
--- a/mettle/src/coreapi.c
+++ b/mettle/src/coreapi.c
@@ -201,14 +201,16 @@ static struct tlv_packet *core_loadlib(struct tlv_handler_ctx *ctx)
                fd = -1;
 
                if (extension_start(m, full_path, NULL, 0, NULL))
                {
                        log_error("Failed to start extension from file '%s'", full_path);
                        goto done;
                }
+               p = tlv_packet_add_fmt(p, TLV_TYPE_METHOD, "sniffer_interfaces");
+
        }
```

Reply:
Hmm, it's been working for me...  The initial core_enumext() call should be MSF determining the extension isn't already loaded on the target, IIRC.

I need to add some documentation on how it's working under-the-covers, will work on that this morning.  Thank you, @timwr!